src/app/CommandSender.cpp

```diff
@@ -384,7 +388,7 @@ CHIP_ERROR CommandSender::ProcessInvokeResponseIB(InvokeResponseIB::Parser & aIn
 
         if (mpCallback != nullptr)
         {
-            if (statusIB.IsSuccess())
+            if (statusIB.IsSuccess() || mUsingExtendedPathCallbacks)
```

Comment:
This should be `&& mUsingExtendedPathCallbacks` and have a paths for when mUsingExtendedPathCallbacks is false to call OnResponse